### DISCUSSION TITLE
Encoding a video to mp4 (ffmpeg) does not play on TV, but remuxing the same streams again to mp4 (ffmpeg) does

### DISCUSSION THREAD

**Discussion by wrzlpete** [staxrip/staxrip#1415] [Q&A]
When I convert an output from davinci resolve that resides inside an mxf container (DNxHR HQX 10bit) to x265 using a container of "ffmpeg mp4" (because the mp4box container result prevents seeking when playing the file on an LG TV internal player app) the file is not playable by the TV at all. But if I remux the same file to "ffmpeg mp4", it is playable. 

I'd love to not have to remux the result just so it works. Is there anything I can change on the initial conversion to fix this? Here's the MediaInfo data from the 2 clips:
[encoded file.MediaInfo.txt](https://github.com/user-attachments/files/16074270/encoded.file.MediaInfo.txt)
[remuxed file.MediaInfo.txt](https://github.com/user-attachments/files/16074269/remuxed.file.MediaInfo.txt)

I see that "Frame rate mode" gets changed to "Variable" when remuxing (what I never actively selected or wanted) but since the underlying video stream is exactly the same I don't see how this changes when remuxing...

**Comment by Dendraspis** [comment]
Maybe the variable frame rate causes the issue.  Hard to tell...
But after encoding with `x265` the only possibility to get a `VFR` result is to use/copy the timestamps from the source file. Make sure, that they are not extracted and/or not included in the container options.

**Comment by wrzlpete** [comment]
Hey, thanks for your reply! 

If I encode the original constant framerate mxf clip to x265 straight into a mkv container, it is then playable on the TV (but not seekable). The .mkv is also the identical constant framerate. 
If I remux this mkv file to a mp4box mp4 container, it has constant framrate, is playable, but not seekable. 
If I remux the mkv file to a ffmpeg mp4 container, it creates the variable framerate and is playable and seekable.
To make things even more confusing, some other constant framerate files work fine, are playable and seekable.
I guess the whole TV playable / seekable thing is too much of a LG TV related quirk to find the cause of in the scope of staxrips options.

But I still do not see how the ffmpeg mp4 remuxing creates a variable framerate file from a constant framerate one. As I can't set any values for any ffmpeg container in staxrip, I cannot check if any timestamps are copied in the process.

**Reply by Dendraspis** [reply]
Maybe it just creates timestamps which is enough to make it variable.

Nevertheless an MKV container should be by default seekable, so I believe this is a hardware related issue. It seems your TV does not support MKV properly or just limited.

You can try to use the muxer that works on your TV at first please in StaxRip, but that's it I guess.

**Reply by wrzlpete** [reply]
Yeah, the documentation on it is pretty scarce. I also just found out that the mkv-seekable-part seems to be an issue that's not necessarily the TV's fault, since it works on an .mkv if it is on a usb drive connected to the tv (should have tested that earlier). 

That adding timestamps is enough to create a variable framerate is good to know, I thought that the stream itself would be altered if it is changed from constant to variable. 

Thanks for the help!
